### PR TITLE
Rejected UI and logic

### PR DESCRIPTION
**Pull request by T-Damer**
- Refactored the `PostIdsStatuses` store to save all the statuses + last user post with all the data
- Last user post is saved after the mint is completed with status `pending`

### CODE REVIEW

File: src/components/BlockchainList/PostsList.tsx [hunk 1]
```diff
@@ -12,9 +12,7 @@ function BlockchainPostsListSuspended() {
 
   return (
     <>
-      {!selectedPosts.length ? (
-        <NoPosts />
-      ) : (
+      {selectedPosts.length ? (
```
Comment (smyzin) [inline src/components/BlockchainList/PostsList.tsx:15]: I guess it's `not necessary changes`. But I'm fine, to match code style

File: src/data.ts [hunk 2]
```diff
@@ -17,3 +17,5 @@ export default {
     twitter: 'SealCredNFT',
   },
 }
+
+export type DataContractNames = keyof typeof data
```
Comment (smyzin) [inline src/data.ts:21]: Maybe we should put it in the `models`? As far as I know, we use it separately from `data` in some places

File: src/components/PostProcessing/index.tsx [hunk 3]
```diff
@@ -15,50 +14,40 @@ import classnames, {
...
-
-  const statusComponent = lastProcessedStatusId ? (
-    <LargeText>Tweet successful</LargeText>
-  ) : (
-    <>
-      <ProcessHeader />
-      <Loading />
-    </>
-  )
+  console.log('lastUserPost', lastUserPost)
```
Comment (smyzin) [inline src/components/PostProcessing/index.tsx:32]: I don't think we need a `console` here

File: src/components/PostProcessing/index.tsx [hunk 4]
```diff
@@ -15,50 +14,40 @@ import classnames, {
...
+        lastUserPost?.status === PostStatus.pending,
+        !lastUserPost
+      )}
+    >
+      {lastUserPost?.status === PostStatus.published ? (
+        <TweetSuccessful tweetId={lastUserPost.tweetId} />
+      ) : lastUserPost?.status === PostStatus.rejected ? (
+        <PostRejected
+          store={lastUserPost.store}
+          blockchainId={lastUserPost.blockchainId}
```
Comment (smyzin) [inline src/components/PostProcessing/index.tsx:46]: May we turn it un a separate component that will handle status and return the right component? Via `switch` for instance or smt like that. Like:
```
case 'pending':
  return <PendingPost pendingPost={lastUserPost} />
case 'rejected':
  return <RejectedPost ../>
case 'publish':
  return <SuccessfulPost ... />
```

File: src/components/PostProcessing/index.tsx [hunk 5]
```diff
@@ -15,50 +14,40 @@ import classnames, {
...
         />
-      ) : null}
+      ) : (
+        <PendingPost pendingPost={lastUserPost} />
```
Comment (smyzin) [inline src/components/PostProcessing/index.tsx:49]: I like the idea of how you separate components by their status 👍

File: src/stores/PostIdsStatuses.ts [hunk 6]
```diff
@@ -1,75 +1,75 @@
...
 
-  for (const { tweetId, status, statusId } of updatedStatuses) {
-    postStatusStore.statuses[name][tweetId] = Promise.resolve({
+  for (const { blockchainId, status, tweetId } of updatedStatuses) {
+    postStatusStore.statuses[name][blockchainId] = Promise.resolve({
       status,
-      statusId,
+      tweetId,
     })
+    console.log('status updated higher', blockchainId, status, tweetId)
```
Comment (smyzin) [inline src/stores/PostIdsStatuses.ts:50]: It's better to remove it

File: src/stores/PostIdsStatuses.ts [hunk 7]
```diff
@@ -1,75 +1,75 @@
...
+    postStatusStore.lastUserPost = {
+      store: name,
+      status,
+      blockchainId,
+      tweetId,
     }
+    console.log(
+      'status updated now last processed post is: ',
+      postStatusStore.lastUserPost
+    )
```
Comment (smyzin) [inline src/stores/PostIdsStatuses.ts:63]: That too

**Review by smyzin — COMMENTED** [review]
It only left to clean the mess up. Otherwise looks pretty good

File: src/components/BlockchainList/PostsList.tsx (hunk 1 above)
Comment (T-Damer) [reply src/components/BlockchainList/PostsList.tsx:15]: Well, this is a little more optimized, because you don't need to do `!` + better readability

File: src/data.ts (hunk 2 above)
Comment (T-Damer) [reply src/data.ts:21]: Okay, let's keep it the same

File: src/components/PostProcessing/index.tsx (hunk 4 above)
Comment (T-Damer) [reply src/components/PostProcessing/index.tsx:46]: Yeah, I tried that, but didn't work, now I almost finished refactoring and tried again, looks like working :)

File: src/components/PostProcessing/TweetSuccessful.tsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { LargeText } from 'components/Text'
+import ViewTweetButton from 'components/PostProcessing/ViewTweetButton'
+
+export default function ({ tweetId }: { tweetId?: number }) {
+  return (
+    <>
+      <LargeText>Tweet successful</LargeText>
+      <ViewTweetButton
+        url={`https://twitter.com/SealCredEmail/status/${tweetId}`}
```
Comment (T-Damer) [inline src/components/PostProcessing/TweetSuccessful.tsx:9]: Oh wait, it should point to exact twitter, I'll fix it

File: src/components/BlockchainList/TwitterLink.tsx [hunk 9]
```diff
@@ -5,33 +5,33 @@ import Delimiter from 'components/Delimiter'
...
 interface TwitterLinkProps {
-  id: number
+  blockchainId: number
 }
 
-function TwitterLinkSuspended({ id }: TwitterLinkProps) {
-  const { currentStatuses } = useSnapshot(postIdsStatuses)
-  const statusId = currentStatuses[id]?.statusId
+function TwitterLinkSuspended({ blockchainId }: TwitterLinkProps) {
+  const currentStatuses = useSnapshot(postIdsStatuses.currentStatuses)
```
Comment (smyzin) [inline src/components/BlockchainList/TwitterLink.tsx:12]: Why did you change it from `const { currentStatuses } = useSnapshot(postIdsStatuses)`?

**Review by smyzin — APPROVED** [review]
Lgtm

File: src/stores/PostIdsStatuses.ts [hunk 10]
```diff
@@ -88,26 +95,31 @@ subscribeKey(PostStore, 'selectedPosts', updateStatusesForSelectedPosts)
...
-  for (const name in postStatusStore.processing) {
+  for (const name in Object.keys(data)) {
+    if (!postStatusStore.statuses[name]) return
+    const ids: number[] = []
+
+    await Promise.all(
+      Object.entries(postStatusStore.statuses[name]).map(
+        async ([blockchainId, postData]) => {
+          const { status } = await postData
+          if (status === PostStatus.pending || status === PostStatus.approved)
```
Comment (T-Damer) [inline src/stores/PostIdsStatuses.ts:106]: adds only pending and approved posts into ids that should be checked, because we shouldn't check rejected or published, those statuses can't be changed anymore

File: src/components/CreatePost/Form.tsx [hunk 11]
```diff
@@ -110,9 +114,22 @@ export default function () {
                     ...posts,
                   ])
 
-                  PostIdsStatuses.processing[ledgerType].add(id.toNumber())
+                  if (PostIdsStatuses.lastUserPost)
```
Comment (T-Damer) [inline src/components/CreatePost/Form.tsx:117]: An entry point for `lastUserPost`

**Review by backmeupplz — APPROVED** [review]
